Create folder automatically for TestNG suite file

Case:
```
new TestNGEngine(
                new CachedTestNGSuite(
                        new LoadableTestNGSuite(
                                new FileSystemOfClasspathClasses(),
                                "sdd",
                                new RegexCondition("^com.extsoft.comments.tests(.+)?")
                        )
                )
        ).run();
```
Actual: 
```
Sunshine found 1 classes by the specified pattern. They all will be passed to appropriate xUnit engine.
Classes:
- com.extsoft.comments.tests.AutomatedTests
Exception in thread "main" io.github.tatools.sunshine.core.EngineException: Some problem occurs in the TestNG engine
	at io.github.tatools.sunshine.testng.TestNGEngine.run(TestNGEngine.java:47)
	at com.extsoft.comments.App.main(App.java:31)
Caused by: io.github.tatools.sunshine.core.SuiteException: java.nio.file.NoSuchFileException: sdd/sunshine-suite.xml
	at io.github.tatools.sunshine.testng.LoadableTestNGSuite.tests(LoadableTestNGSuite.java:49)
	at io.github.tatools.sunshine.testng.LoadableTestNGSuite.tests(LoadableTestNGSuite.java:14)
	at io.github.tatools.sunshine.testng.CachedTestNGSuite.tests(CachedTestNGSuite.java:26)
	at io.github.tatools.sunshine.testng.TestNGEngine.run(TestNGEngine.java:43)
	... 1 more
Caused by: java.nio.file.NoSuchFileException: sdd/sunshine-suite.xml
	at sun.nio.fs.UnixException.translateToIOException(UnixException.java:86)
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107)
	at sun.nio.fs.UnixFileSystemProvider.newByteChannel(UnixFileSystemProvider.java:214)
	at java.nio.file.spi.FileSystemProvider.newOutputStream(FileSystemProvider.java:434)
	at java.nio.file.Files.newOutputStream(Files.java:216)
	at java.nio.file.Files.write(Files.java:3292)
	at io.github.tatools.sunshine.core.FileBase.write(FileBase.java:47)
	at io.github.tatools.sunshine.testng.LoadableTestNGSuite.tests(LoadableTestNGSuite.java:46)
	... 4 more
```
Expected: 
`sdd` will be created if it doesn't exist.

--- a/sunshine-testng/src/main/java/io/github/tatools/sunshine/testng/LoadableTestNGSuite.java
+++ b/sunshine-testng/src/main/java/io/github/tatools/sunshine/testng/LoadableTestNGSuite.java
@@ -35,16 +35,32 @@ public final class LoadableTestNGSuite implements TestNGSuite {
         );
     }
 
-    public LoadableTestNGSuite(FileSystem fileSystem, Directory suitePath, Condition filter) {
+    /**
+     * Constructs new instance with the specified file system, suite's directory and tests filter. All filtered tests
+     * will be printed to {@link System#out}.
+     *
+     * @param fileSystem        the place with the tests
+     * @param xmlSuiteDirectory the place to store suite file
+     * @param filter            the filter to be used to select desired tests
+     * @since 0.1
+     */
+    public LoadableTestNGSuite(FileSystem fileSystem, Directory xmlSuiteDirectory, Condition filter) {
         this(
                 new SunshineSuitePrintable(new SunshineSuiteFilterable(new SunshineSuiteBase(fileSystem), filter)),
-                suitePath
+                xmlSuiteDirectory
         );
     }
 
-    public LoadableTestNGSuite(SunshineSuite artifacts, Directory suitePath) {
-        this.artifacts = artifacts;
-        this.suitePath = suitePath;
+    /**
+     * Constructs new instance with the specified suite and suite's directory.
+     *
+     * @param suite             the tests to be used
+     * @param xmlSuiteDirectory the directory to store suite file
+     * @since 0.1
+     */
+    public LoadableTestNGSuite(SunshineSuite suite, Directory xmlSuiteDirectory) {
+        artifacts = suite;
+        suitePath = xmlSuiteDirectory;
     }
 
     @Override
@@ -52,12 +68,12 @@ public final class LoadableTestNGSuite implements TestNGSuite {
         XmlSuite xmlSuite = new XmlSuite();
         xmlSuite.setName("Sunshine suite");
         try {
-            for (SunshineTest sunshineTest : this.artifacts.tests()) {
+            for (SunshineTest sunshineTest : artifacts.tests()) {
                 XmlTest test = new TestNGTest(sunshineTest).object();
                 test.setSuite(xmlSuite);
                 xmlSuite.addTest(test);
             }
-            FileBase fileBase = new FileBase(this.suitePath, "sunshine-suite.xml");
+            FileBase fileBase = new FileBase(suitePath, "sunshine-suite.xml");
             fileBase.write(xmlSuite.toXml());
             return fileBase;
         } catch (TestException | IOException e) {

--- a/sunshine-testng/src/main/java/io/github/tatools/sunshine/testng/LoadableTestNGSuite.java
+++ b/sunshine-testng/src/main/java/io/github/tatools/sunshine/testng/LoadableTestNGSuite.java
@@ -17,7 +17,7 @@ import java.io.IOException;
 public final class LoadableTestNGSuite implements TestNGSuite {
 
     private final SunshineSuite artifacts;
-    private final Directory suitePath;
+    private final File suiteXml;
 
     /**
      * Construct new instance with the specified file system, suite's directory and tests filter.
@@ -61,8 +61,8 @@ public final class LoadableTestNGSuite implements TestNGSuite {
      * @since 0.1
      */
     public LoadableTestNGSuite(SunshineSuite suite, Directory xmlSuiteDirectory) {
-        this.artifacts = suite;
-        this.suitePath = xmlSuiteDirectory;
+        artifacts = suite;
+        suiteXml = new FileBase(xmlSuiteDirectory, "sunshine-suite.xml");
     }
 
     @Override
@@ -75,9 +75,8 @@ public final class LoadableTestNGSuite implements TestNGSuite {
                 test.setSuite(xmlSuite);
                 xmlSuite.addTest(test);
             }
-            FileBase fileBase = new FileBase(this.suitePath, "sunshine-suite.xml");
-            fileBase.write(xmlSuite.toXml());
-            return fileBase;
+            suiteXml.write(xmlSuite.toXml());
+            return suiteXml;
         } catch (TestException | IOException e) {
             throw new SuiteException(e);
         }

--- a/sunshine-testng/src/main/java/io/github/tatools/sunshine/testng/LoadableTestNGSuite.java
+++ b/sunshine-testng/src/main/java/io/github/tatools/sunshine/testng/LoadableTestNGSuite.java
@@ -11,13 +11,28 @@ import java.io.IOException;
  * @version $Id$
  * @since 0.1
  */
+@SuppressWarnings("WeakerAccess")
 public final class LoadableTestNGSuite implements TestNGSuite {
 
     private final SunshineSuite artifacts;
     private final Directory suitePath;
 
-    public LoadableTestNGSuite(FileSystem fileSystem, String suitePath, Condition filter) {
-        this(fileSystem, new DirectoryBase(suitePath), filter);
+    /**
+     * Construct new instance with the specified file system, suite's directory and tests filter.
+     * If suite's directory ({@code xmlSuiteDirectory}) doesn't exist, it will be created automatically.
+     *
+     * @param fileSystem        the place with the tests
+     * @param xmlSuiteDirectory the place to store suite file
+     * @param filter            the filter to be used to select desired tests
+     * @see #LoadableTestNGSuite(FileSystem, Directory, Condition)
+     * @since 0.2
+     */
+    public LoadableTestNGSuite(FileSystem fileSystem, String xmlSuiteDirectory, Condition filter) {
+        this(
+                fileSystem,
+                new DirectoryWithAutomaticCreation(new DirectorySafe(new DirectoryBase(xmlSuiteDirectory))),
+                filter
+        );
     }
 
     public LoadableTestNGSuite(FileSystem fileSystem, Directory suitePath, Condition filter) {
@@ -37,12 +52,12 @@ public final class LoadableTestNGSuite implements TestNGSuite {
         XmlSuite xmlSuite = new XmlSuite();
         xmlSuite.setName("Sunshine suite");
         try {
-            for (SunshineTest sunshineTest : artifacts.tests()) {
+            for (SunshineTest sunshineTest : this.artifacts.tests()) {
                 XmlTest test = new TestNGTest(sunshineTest).object();
                 test.setSuite(xmlSuite);
                 xmlSuite.addTest(test);
             }
-            FileBase fileBase = new FileBase(suitePath, "sunshine-suite.xml");
+            FileBase fileBase = new FileBase(this.suitePath, "sunshine-suite.xml");
             fileBase.write(xmlSuite.toXml());
             return fileBase;
         } catch (TestException | IOException e) {

--- a/sunshine-testng/src/main/java/io/github/tatools/sunshine/testng/LoadableTestNGSuite.java
+++ b/sunshine-testng/src/main/java/io/github/tatools/sunshine/testng/LoadableTestNGSuite.java
@@ -7,6 +7,8 @@ import org.testng.xml.XmlTest;
 import java.io.IOException;
 
 /**
+ * The {@link LoadableTestNGSuite} class represents a TestNG suite prepared from the Java classes.
+ *
  * @author Dmytro Serdiuk (dmytro.serdiuk@gmail.com)
  * @version $Id$
  * @since 0.1
@@ -59,8 +61,8 @@ public final class LoadableTestNGSuite implements TestNGSuite {
      * @since 0.1
      */
     public LoadableTestNGSuite(SunshineSuite suite, Directory xmlSuiteDirectory) {
-        artifacts = suite;
-        suitePath = xmlSuiteDirectory;
+        this.artifacts = suite;
+        this.suitePath = xmlSuiteDirectory;
     }
 
     @Override
@@ -68,12 +70,12 @@ public final class LoadableTestNGSuite implements TestNGSuite {
         XmlSuite xmlSuite = new XmlSuite();
         xmlSuite.setName("Sunshine suite");
         try {
-            for (SunshineTest sunshineTest : artifacts.tests()) {
+            for (SunshineTest sunshineTest : this.artifacts.tests()) {
                 XmlTest test = new TestNGTest(sunshineTest).object();
                 test.setSuite(xmlSuite);
                 xmlSuite.addTest(test);
             }
-            FileBase fileBase = new FileBase(suitePath, "sunshine-suite.xml");
+            FileBase fileBase = new FileBase(this.suitePath, "sunshine-suite.xml");
             fileBase.write(xmlSuite.toXml());
             return fileBase;
         } catch (TestException | IOException e) {

--- a/sunshine-testng/src/test/java/io/github/tatools/sunshine/testng/LoadableTestNGSuiteTest.java
+++ b/sunshine-testng/src/test/java/io/github/tatools/sunshine/testng/LoadableTestNGSuiteTest.java
@@ -1,0 +1,30 @@
+package io.github.tatools.sunshine.testng;
+
+import io.github.tatools.sunshine.core.Condition.Fake;
+import io.github.tatools.sunshine.core.SuiteException;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.util.ArrayList;
+
+
+/**
+ * @author Dmytro Serdiuk (dmytro.serdiuk@gmail.com)
+ * @version $Id$
+ * @since 0.2
+ */
+public class LoadableTestNGSuiteTest {
+
+    @Rule
+    public TemporaryFolder testFolder = new TemporaryFolder();
+
+    @Test
+    public void testAutomaticSuiteDirectoryCreation() throws SuiteException {
+        new LoadableTestNGSuite(
+                ArrayList::new,
+                this.testFolder.getRoot().getAbsolutePath() + "/custom",
+                new Fake(true)
+        ).tests();
+    }
+}

